### PR TITLE
Meiki server cleanup apr 2022

### PR DESCRIPTION
**Pull request by RainingComputers**

### CODE REVIEW

File: meiki_server/auth/routes.go [hunk 1]
```diff
@@ -74,7 +79,12 @@ func getCreateHandler(ctx context.Context, a Auth) gin.HandlerFunc {
 func getDeleteHandler(ctx context.Context, a Auth) gin.HandlerFunc {
 	return func(c *gin.Context) {
 		var creds Credentials
-		c.BindJSON(&creds)
+		err := c.BindJSON(&creds)
+
+		if err != nil {
```
Comment (alexsanjoseph) [inline meiki_server/auth/routes.go:65]: some drying opportunity here?

Can we add the parseCredential as a(nother) middleware instead? (thoughts for a different PR)

File: meiki_server/auth/routes.go [hunk 2]
```diff
@@ -218,7 +232,7 @@ func getAuthStatus(ctx context.Context, a Auth) gin.HandlerFunc {
 
 func CreateRoutes(router *gin.RouterGroup, ctx context.Context, auth Auth) {
 	router.POST("/create", getCreateHandler(ctx, auth))
-	router.POST("/delete", getDeleteHandler(ctx, auth))
+	router.DELETE("/delete", getDeleteHandler(ctx, auth))
```
Comment (alexsanjoseph) [inline meiki_server/auth/routes.go]: 😄

File: meiki_server/main.go [hunk 3]
```diff
@@ -22,21 +22,21 @@ func run() error {
...
-		panic("unable to connect to mongo for test suite")
+		log.Error("Unable to connect to mongo", zap.Error(err))
+		return err
 	}
 
-	meiki_db := client.Database("meiki")
-	userColl := meiki_db.Collection("users")
-	tokenColl := meiki_db.Collection("tokens")
-	notesColl := meiki_db.Collection("notes")
+	meikiDb := client.Database("meiki")
```
Comment (alexsanjoseph) [inline meiki_server/main.go]: meikiDB instead?

example (https://pkg.go.dev/database/sql)

File: meiki_server/auth/routes.go (hunk 1 above)
Comment (RainingComputers) [reply meiki_server/auth/routes.go:65]: Hmm, yep, will look into it

File: meiki_server/main.go (hunk 3 above)
Comment (RainingComputers) [reply meiki_server/main.go]: yep

File: meiki_server/notes/notes_test.go [hunk 4]
```diff
@@ -75,15 +75,25 @@ var note3 = notes.Note{
 	Content:  "What is my purpose?; You are a dummy for this test; Oh my god",
 }
 
+var invalidNote = notes.Note{
```
Comment (alexsanjoseph) [inline meiki_server/notes/notes_test.go:78]: Why is this variable out in the global scope (I'm missing something)

File: meiki_server/notes/routes.go [hunk 5]
```diff
@@ -7,18 +7,45 @@ import (
...
+const MSG_INVALID_ID = "Invalid id"
+const MSG_UNABLE_TO_PERFORM_ACTION = "Unable perform requested action"
 
 type CreateRequest struct {
-	Title    string `json:"title"`
-	UserName string `json:"username"`
+	Title string `json:"title"`
 }
 
+type RenameRequest = CreateRequest
```
Comment (alexsanjoseph) [inline meiki_server/notes/routes.go]: I think the abstraction is misleading here. At best both of this should a NewRequest type

File: meiki_server/notes/routes.go [hunk 6]
```diff
@@ -144,5 +175,6 @@ func CreateRoutes(router *gin.RouterGroup, ctx context.Context, ns NotesStore) {
 	router.GET("/list", getListHandler(ctx, ns))
 	router.GET("/read/:id", getReadHandler(ctx, ns))
 	router.POST("/update/:id", getUpdateHandler(ctx, ns))
-	router.POST("/delete/:id", getDeleteHandler(ctx, ns))
+	router.POST("/rename/:id", genRenameHandler(ctx, ns))
```
Comment (alexsanjoseph) [inline meiki_server/notes/routes.go]: update/rename should be PUT, I think (https://www.mscharhag.com/api-design/http-post-put-patch)

**Review by alexsanjoseph — APPROVED** [review]
Nice PR, thanks!

File: meiki_server/notes/notes_test.go (hunk 4 above)
Comment (RainingComputers) [reply meiki_server/notes/notes_test.go:78]: should have been `const`, these are fixtures

File: meiki_server/notes/routes.go (hunk 5 above)
Comment (RainingComputers) [reply meiki_server/notes/routes.go]: `TitleRequest`?

File: meiki_server/notes/routes.go (hunk 6 above)
Comment (RainingComputers) [reply meiki_server/notes/routes.go]: yep

File: meiki_server/notes/routes.go (hunk 5 above)
Comment (alexsanjoseph) [reply meiki_server/notes/routes.go]: yeah that works

File: meiki_server/notes/notes_test.go [hunk 7]
```diff
@@ -75,15 +75,25 @@ var note3 = notes.Note{
 	Content:  "What is my purpose?; You are a dummy for this test; Oh my god",
```
Comment (alexsanjoseph) [inline meiki_server/notes/notes_test.go:75]: Oh I missed this - R&M reference?

Comment (RainingComputers) [reply meiki_server/notes/notes_test.go:75]: yep :P

File: meiki_server/notes/notes_test.go (hunk 4 above)
Comment (RainingComputers) [reply meiki_server/notes/notes_test.go:78]: Only primitive types can be a global const (https://stackoverflow.com/questions/38542588/making-constant-global-variables-of-a-custom-struct-type), so seems like we ate stuck with var here :(

File: meiki_server/auth/routes.go (hunk 1 above)
Comment (RainingComputers) [reply meiki_server/auth/routes.go:65]: Added as a TODO to be addressed in another PR

**Pull request merged**